docs/details/tests.md

```diff
@@ -32,3 +32,113 @@ The command `memtest` uses this idea to find memory leaks in pure functions. How
 The return code is equal to the amount of functions with memory leaks.
 
 If the function fails, the command will ignore it and still test the function for leaks. Side-effects shouldn't happen unconditionally, even if the function fails. If you want to find unexpected failures, use `test` command instead.
+
+## Constant value for arguments
+
+The `deal.cases` constructor accepts `kwargs` argument where you can specify constant values for the function arguments. For example:
+
+```python
+@deal.raises(ZeroDivisionError)
+def div(a: int, b: int) -> float:
+    assert a == 1
+    return a / b
+
+# Every test case calls `div` function with `a=1`.
+# So, random values are generated only for `b`.
+@deal.cases(div, kwargs=dict(a=1))
+def test_div(case):
+    case()
+```
+
+## Custom strategies
+
+Under the hood, `deal.cases` uses [hypothesis](https://hypothesis.readthedocs.io/en/latest/index.html) testing framework to generate test cases. The trick is that `kwargs` argument of `deal.cases` can contain hypothesis strategies:
+
+```python
+import hypothesis.strategies as st
+
+@deal.raises(ZeroDivisionError)
+def div(a: int, b: int) -> float:
+    assert a >= 10
+    return a / b
+
+cases = deal.cases(
+    func=div,
+    kwargs=dict(
```

Comment:
I know that this is intended, but why `dict()` and not `{}`?

Reply:
1. To make it look like you would directly call the function.

    ```python
    f(a=1)
    # looks-alike:
    deal.cases(f, kwargs=dict(a=1))
    # less similar:
    deal.cases(f, kwargs={'a': 1})
    ```
2. To have the check that all names are valid identifiers at parsing time rather than runtime:

    ```python
    # syntax error:
    deal.cases(f, kwargs=dict(13=14))
    # will fail only in runtime:
    deal.cases(f, kwargs={13: 14})
    ```